src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -696,14 +819,803 @@ public boolean validatePortRange(Integer portNumber) {
+				}
+				break;
+			case CANCEL:
+				this.cachedDecoderConfigs.clear();
+				this.cachedDecoderConfigs.addAll(this.realtimeDecoderConfigs);
+				populateDecoderControl(stats, advancedControllableProperties, decoderID);
+				break;
+			default:
+				if (logger.isWarnEnabled()) {
+					logger.warn(String.format("Operation %s with value %s is not supported.", controllableProperty, value));
+				}
+				throw new IllegalStateException(String.format("Operation %s with value %s is not supported.", controllableProperty, value));
+		}
+	}
+
+	/**
+	 * This method is used for populate local extended stats when emergency delivery:
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 */
+	private void populateLocalExtendedStats(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		this.localExtendedStatistics.setStatistics(stats);
+		this.localExtendedStatistics.setControllableProperties(advancedControllableProperties);
+		isEmergencyDelivery = true;
+	}
+
+	/**
+	 * This method is used to perform decoder SDI control: update
+	 *
+	 * @param decoderConfig set of decoder config info
+	 * @param decoderID is ID of decoder
+	 * @return String response
+	 */
+	private String performDecoderSDIUpdateControl(DecoderConfig decoderConfig, Integer decoderID) {
+		try {
+			String request = decoderConfig.contributeCommand(CommandOperation.OPERATION_VIDDEC.getName(), decoderID, CommandOperation.SET.getName());
+			String response = send(request);
+			if (!StringUtils.isNullOrEmpty(response)) {
+				return response;
+			}
+		} catch (Exception e) {
+			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + DecoderConstant.NEXT_LINE + e.getMessage(), e);
+		}
+		return DecoderConstant.EMPTY;
+	}
+
+	/**
+	 * This method is used to perform decoder SDI control: start/ stop
+	 *
+	 * @param decoderID is ID of decoder
+	 * @param active start/ stop activation
+	 * @return Boolean control result
+	 */
+	private boolean performActiveDecoderSDIControl(DecoderConfig decoderConfig, Integer decoderID, String active) {
+		try {
+			String request = CommandOperation.OPERATION_VIDDEC.getName()
+					.concat(DecoderConstant.SPACE)
+					.concat(decoderID.toString())
+					.concat(DecoderConstant.SPACE)
+					.concat(active);
+			String response = send(request);
+			return !StringUtils.isNullOrEmpty(response) && response.contains(DecoderConstant.SUCCESSFUL_RESPONSE);
+		} catch (Exception e) {
+			reverseActiveControlAfterControlFailed(decoderConfig, decoderID);
+			throw new ResourceNotReachableException(DecoderConstant.DECODER_CONTROL_ERR + active + DecoderConstant.NEXT_LINE + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * This method is used to reverse active control value after control failed
+	 *
+	 * @param decoderConfig list of decoder data
```

Comment:
Add param decoderID in javadoc. Also check missing @param/return/throws  for other methods.

Reply:
I fixed it